### PR TITLE
Add introspection via pg_catalog

### PR DESCRIPTION
**Pull request by aljazerzen**
Uses underlying pg_catalog and joins-in our schema to export nice table and columns names.

Ref #4814

### CODE REVIEW

File: edb/pgsql/metaschema.py [hunk 1]
```diff
@@ -5049,169 +5046,98 @@ def _generate_sql_information_schema() -> List[dbops.View]:
...
+    # A few tables in here were causing problems, so let's hide them as an
+    # implementation detail.
+    # To be more specific:
+    # - following tables were missing from information_schema:
+    #   Link.properties, ObjectType.links, ObjectType.properties
+    # - even though introspection worked, I wasn't able to select from some
+    #   tables in cfg and sys
+
+    # For making up oids of schemas that represent modules
+    uuid_to_oid = dbops.Function(
```
Comment (msullivan) [inline edb/pgsql/metaschema.py:5079]: Hm, what do we think the odds of collision are? How big is oid?

**Review by msullivan — APPROVED** [review]
Seems like a big improvement!

File: edb/pgsql/metaschema.py (hunk 1 above)
Comment (elprans) [reply edb/pgsql/metaschema.py:5079]: 32-bit signed integers, so technically this can overflow.

File: edb/pgsql/metaschema.py [hunk 2]
```diff
@@ -5267,7 +5207,88 @@ def _generate_sql_information_schema() -> List[dbops.View]:
...
+            typowner,
+            typlen,
+            typbyval,
+            typtype,
+            typcategory,
+            typispreferred,
+            typisdefined,
+            typdelim,
+            typrelid,
+            typsubscript,
```
Comment (fantix) [inline edb/pgsql/metaschema.py:5224]: Looks like `typsubscript` is only added in [PostgreSQL 14](https://www.postgresql.org/docs/14/catalog-pg-type.html), thus this breaks [using EdgeDB on PostgreSQL 13](https://github.com/edgedb/edgedb/actions/runs/3992636910/jobs/6857300141), which will still have a (final) release on [Nov 2025](https://www.postgresql.org/support/versioning/). @elprans should we fix the use of `typsubscript` or drop the support of PostgreSQL 13?

Comment (aljazerzen) [reply edb/pgsql/metaschema.py:5224]: We can just remove `typsubscript` from here. If this is a change in recent pg version, no tool will depend on it.

**Pull request merged**